src/lib.rs

```diff
@@ -1641,17 +1640,14 @@ impl<'a, T: Clone> EventListenerFuture for SendInner<'a, T> {
             }
 
             // Sending failed - now start listening for notifications or wait for one.
-            match &mut this.listener {
-                None => {
-                    // Start listening and then try sending again.
-                    let inner = inner.write().unwrap();
-                    this.listener = Some(inner.send_ops.listen());
-                }
-                Some(l) => {
-                    // Wait for a notification.
-                    ready!(strategy.poll(l.as_mut(), context));
-                    this.listener = None;
-                }
+            if this.listener.is_none() {
```

Comment:
Why change from match to if? When we've blocks for both `None` and `Some`, `match` is more readable IMO. Besides, it's irrelevant to the containing commit.